internal/pkg/warmup/warmup.go

```diff
@@ -132,7 +134,11 @@ func (w Warmup) HTTPWarmupWorker(wg *sync.WaitGroup, requests <-chan http.Reques
 	for request := range requests {
 		time.Sleep(time.Duration(requestDelayMilliseconds) * time.Millisecond)
 
-		resp := w.Target.httpClient.SendRequest(request.Method, request.Path, headers, request.Body)
+		headersMap := util.ToHeaders(headers)
+		// Overwrite Content-Encoding header if required
```

Comment:
```suggestion
		// If the Content-Encoding header is already set by the user, this choice will be respected
```

Reply:
> Copy copies all key/value pairs in src adding them to dst. When a key in src is already present in dst, the value in dst will be overwritten by the value associated with the key in src.

So it is not respecting the user input. It overwrites it. Which is fine i think, because when compression is enabled by mittens flag this should be expected. Content-Encoding is only used for compression purposes.